Add support for whitespace inside build args

cybojenix opened the pull request:
Utilise a bash array to expand the parameters correctly for the img command.

We've been facing an issue where if we have a build argument that contains whitespace, `img` will think the second word is the context.

```sh
$ build_args="--build-arg VAR=some var"
$ img build --no-console -t safe-restarter:latest $build_args .
... 
failed to solve: failed to resolve dockerfile: error from sender: failed to resolve : lstat var: no such file or directory
```

Changing to a bash array allows us to do parameter expansion:

```sh
$ build_args=( '--build-arg' 'VAR=some var' )
$ img build --no-console -t safe-restarter:latest "${build_args[@]}" .
# builds as expected with `VAR` set to `some var`
```

vito commented:
makes sense! would you mind signing the commit to make the DCO bot happy?: https://github.com/concourse/builder-task/pull/23/checks?check_run_id=126348364

cybojenix commented:
Hi @vito, I've rebased the PR to signoff the commits.
I've also fixed whitespace when using `build_args_file`.

This introduce a small behaviour change, so `BUILD_ARG_*` and `BUILD_ARGS_FILE` can be used at the same time.
Let me know if I should change it back to the previous behaviour where `BUILD_ARG_*` is ignored when `BUILD_ARS_FILE` is set

vito commented on build:
```diff
@@ -97,25 +95,26 @@ elif [ -n "$TARGET" ]; then
-build_args=""
+build_args=()
+
+while read -r line && test "${line}"; do
```
The `test` here seems to be missing an operator (`-n`?)

cybojenix replied:
this isn't required, however I've added it to be explicit we're testing for string length:

```sh
$ test ""; echo $?
1
$ test "contents"; echo $?
0
```

vito commented on build:
```diff
@@ -97,25 +95,26 @@ elif [ -n "$TARGET" ]; then
 
-  expanded_build_args=()
-  while IFS= read -r line || [ -n "$line" ];
+  while IFS= read -r line && test -n "${line}" || [ -n "$line" ];
```
`test -n "${line}" || [ -n "$line" ]` are both the same thing (`[` is an alias to `test`):

```
alex@wx020:~/workspace/builder-task$ help [
[: [ arg... ]
    Evaluate conditional expression.
    
    This is a synonym for the "test" builtin, but the last argument must
    be a literal `]', to match the opening `['.
```
